src/main/java/org/jenkinsci/plugins/authorizeproject/AuthorizeProjectStrategy.java

```diff
@@ -110,23 +116,97 @@ public AuthorizeProjectStrategyDescriptor getDescriptor() {
      * @throws AccessDeniedException if the current user is not allowed to reconfigure the specified job
      * @since 1.3.0
      */
-    public final void checkConfigurePermission(AccessControlled context) {
-        if (!hasConfigurePermission(context)) {
-            throw new AccessDeniedException(Messages.AuthorizeProjectStrategy_UserNotAuthorized(
+    public final void checkJobConfigurePermission(AccessControlled context) {
+        if (!hasJobConfigurePermission(context)) {
+            throw new AccessDeniedException(Messages.AuthorizeProjectStrategy_UserNotAuthorizedForJob(
                     Jenkins.getAuthentication().getName()
             ));
         }
     }
     
     /**
      * Tests if the job can be reconfigured by the current user when this strategy is the configured strategy.
+     * Users with {@link Jenkins#ADMINISTER} permission skips this check.
      *
      * @param context the context of the job
      * @return {@code true} if and only if the current user is allowed to reconfigure the specified job.
      * @since 1.3.0
      */
-    public boolean hasConfigurePermission(AccessControlled context) {
+    public boolean hasJobConfigurePermission(AccessControlled context) {
         return true;
     }
 
+    /**
+     * Checks that the authorization can be configured by the current user.
+     *
+     * @param context the context of the job
+     * @throws AccessDeniedException if the current user is not allowed to configure this authorization
+     * @since 1.3.0
+     */
+    public final void checkAuthorizationConfigurePermission(AccessControlled context) {
+        if (!hasAuthorizationConfigurePermission(context)) {
+            throw new AccessDeniedException(Messages.AuthorizeProjectStrategy_UserNotAuthorized(
+                    Jenkins.getAuthentication().getName()
+            ));
+        }
+    }
+
+    /**
+     * Tests if the authorization can be configured by the current user.
+     * Users with {@link Jenkins#ADMINISTER} permission skips this check.
+     *
+     * @param context the context of the job
+     * @return {@code true} if and only if the current user is allowed to configure this authorization.
+     * @since 1.3.0
+     */
+    public boolean hasAuthorizationConfigurePermission(AccessControlled context) {
+        return true;
+    }
+
+    /**
+     * If we are being deserialized outside of loading the initial jobs (or reloading) then we need to cross check
+     * the strategy permissions to defend against somebody trying to push a configuration relating to a user other
+     * than themselves.
+     *
+     * @return {@code this}
+     * @throws ObjectStreamException if the object cannot be deserialized.
+     * @see AuthorizeProjectProperty#setStrategyCritical()
+     */
+    private Object readResolve() throws ObjectStreamException {
+        checkUnsecuredConfiguration();

```

Comment:
Moved from `AuthorizeProjectProperty` and extracted into `checkUnsecuredConfiguration` to avoid deep indentations and add `ADMINISTER` check.